internal/nix/input.go

```diff
@@ -20,24 +20,50 @@ import (
 	"go.jetpack.io/devbox/internal/lock"
 )
 
+// Input represents a "package" added to the devbox.json config.
+// The word "input" is used because it will be referenced in a generated flake.nix.
+// A unique feature of flakes is that they have well-defined "inputs" and "outputs".
+// This Input will be aggregated into a specific "flake input" (see shellgen.flakeInput).
 type Input struct {
 	url.URL
 	lockfile lock.Locker
-	Raw      string
+
+	// Raw is the devbox package name from the devbox.json config.
+	// Raw has a few forms:
+	// 1. Devbox Packages
+	//    a. versioned packages
+	//       examples:  go@1.20, python@latest
+	//    b. any others?
+	// 2. Local
+	//    flakes in a relative sub-directory
+	//    example: ./local_flake_subdir#myPackage
+	// 3. Github
+	//    remote flakes with raw name starting with `Github:`
+	//    example: github:nixos/nixpkgs/5233fd2ba76a3accb5aaa999c00509a11fd0793c#hello
+	Raw string
 
 	normalizedPackageAttributePathCache string // memoized value from normalizedPackageAttributePath()
 }
 
-func InputsFromStrings(names []string, l lock.Locker) []*Input {
+// InputsFromStrings constructs Input from the list of package names provided.
+// These names correspond to devbox packages from the devbox.json config.
+func InputsFromStrings(rawNames []string, l lock.Locker) []*Input {
 	inputs := []*Input{}
-	for _, name := range names {
-		inputs = append(inputs, InputFromString(name, l))
+	for _, rawName := range rawNames {
+		inputs = append(inputs, InputFromString(rawName, l))
 	}
 	return inputs
 }
 
+// InputsFromStrings constructs Input from the raw name provided.
+// The raw name corresponds to a devbox package from the devbox.json config.
 func InputFromString(raw string, locker lock.Locker) *Input {
+	// We ignore the error because: TODO why?
```

Comment:
@mikeland73 why do we ignore the error?

Reply:
The input is pretty validated so this would be a very rare failure, but this was mostly to avoid a large cascading change when this struct was introduced.